governator-core/src/main/java/com/netflix/governator/LifecycleManager.java

```diff
@@ -60,17 +62,19 @@ public synchronized void notifyStarted() {
     public synchronized void notifyStartFailed(Throwable t) {
         if (state.compareAndSet(State.Starting, State.Stopped)) {
             failureReason = t;
-            for (LifecycleListener listener : listeners) {
-                listener.onStopped(t);
+            LifecycleListener[] asArray = listeners.toArray(new LifecycleListener[0]);
```

Comment:
didn't see the 'descendingIterator()' part; I'll add that in

On Thu, Jun 2, 2016 at 9:03 AM Tom Cellucci <tcellucci@gmail.com> wrote:

> that would iterate in the wrong order; the array is present to reverse the
> direction.
>
>
> On Thu, Jun 2, 2016 at 8:56 AM elandau <notifications@github.com> wrote:
>
>> In
>> governator-core/src/main/java/com/netflix/governator/LifecycleManager.java
>> <https://github.com/Netflix/governator/pull/299#discussion_r65567408>:
>>
>> > @@ -60,17 +62,19 @@ public synchronized void notifyStarted() {
>> >      public synchronized void notifyStartFailed(Throwable t) {
>> >          if (state.compareAndSet(State.Starting, State.Stopped)) {
>> >              failureReason = t;
>> > -            for (LifecycleListener listener : listeners) {
>> > -                listener.onStopped(t);
>> > +            LifecycleListener[] asArray = listeners.toArray(new LifecycleListener[0]);
>>
>> nit: I think the code would read better if we use an iterator instead of
>> dealing with raw arrays.
>>
>> Iterator<String> iter = new LinkedList<>(listeners).descendingIterator();while (iter.hasNext()) {
>>     iter.next().onStopped();
>> }
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Netflix/governator/pull/299/files/e74ae6d7b160296c4705062f881879efe329a178#r65567408>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/ACIO-4HQFq33aTlDNuWYSxpxO315h9VIks5qHv0QgaJpZM4IsEMl>
>> .
>>
>